testing/test_metal_context.mm

```diff
@@ -71,36 +77,35 @@
 
 TestMetalContext::TextureInfo TestMetalContext::CreateMetalTexture(const SkISize& size) {
   std::scoped_lock lock(textures_mutex_);
-  auto texture_descriptor = fml::scoped_nsobject{
-      [[MTLTextureDescriptor texture2DDescriptorWithPixelFormat:MTLPixelFormatBGRA8Unorm
-                                                          width:size.width()
-                                                         height:size.height()
-                                                      mipmapped:NO] retain]};
+  MTLTextureDescriptor* texture_descriptor =
+      [MTLTextureDescriptor texture2DDescriptorWithPixelFormat:MTLPixelFormatBGRA8Unorm
+                                                         width:size.width()
+                                                        height:size.height()
+                                                     mipmapped:NO];
 
   // The most pessimistic option and disables all optimizations but allows tests
   // the most flexible access to the surface. They may read and write to the
   // surface from shaders or use as a pixel view.
-  texture_descriptor.get().usage = MTLTextureUsageUnknown;
+  texture_descriptor.usage = MTLTextureUsageUnknown;
 
   if (!texture_descriptor) {
     FML_CHECK(false) << "Invalid texture descriptor.";
     return {.texture_id = -1, .texture = nullptr};
   }
 
   id<MTLDevice> device = (__bridge id<MTLDevice>)GetMetalDevice();
-  sk_cfp<void*> texture = sk_cfp<void*>{[device newTextureWithDescriptor:texture_descriptor.get()]};
-
+  id<MTLTexture> texture = [device newTextureWithDescriptor:texture_descriptor];
```

Comment:
Is this a retain sink (and annotated as such)? Or was the old code leaking?

Reply:
This `sk_cfp<void*>` ctor retains the texture. I've just migrated that retain down a few lines to (new) line 104.

Reply:
Oh wait -- if it's using CFRetain, it's insufficient for us to just bridge it. 

We need to __bridge_retained to take it out of ARC management, but then we want to hand that retained pointer to the sk_cfp in a way that doesn't cause it to be retained _again_ by the CFRetain in its ctor, so I think what we really want is to pass it in via `reset`.

I mean what we really want is to not have to deal with this sort of thing at all. If I can migrate the other fields to Obj-C fields on the C++ class, then maybe doing the same here and keeping them in an NSDictionary is the right way.

Reply:
An ugly hack might be to have the public header hold a single `void*  objc_` field that holds all the objc fields (`device`, `commandQueue`, and a `textures` dictionary), but I'm not sure it's sufficiently less ugly than the current code.

Reply:
sgtm. I'm going to land as-is to keep this one small then send a quick refactor. I'm also just going to do the right thing and do a full refactor of embedder test based off the  experimental Metal-only refactor I've been doing, which has a ton of other benefits in terms of code cleanliness.